src/object/cli_obj.c

```diff
@@ -3446,12 +3446,14 @@ obj_comp_cb(tse_task_t *task, void *data)
 			}
 		} else {
 			if (obj_auxi->is_ec_obj && task->dt_result == 0 &&
-			    obj_auxi->opc == DAOS_OBJ_RPC_FETCH &&
-			    obj_auxi->bulks != NULL) {
+			    obj_auxi->opc == DAOS_OBJ_RPC_FETCH) {
 				daos_obj_fetch_t *args = dc_task_get_args(task);
 
-				obj_ec_fetch_set_sgl(&obj_auxi->reasb_req,
-						     args->nr);
+				if (obj_auxi->bulks != NULL)
+					obj_ec_fetch_set_sgl(
+						&obj_auxi->reasb_req, args->nr);
+				obj_ec_update_iod_size(&obj_auxi->reasb_req,
+						       args->nr);
```

Comment:
seems in obj_size_fetch_cb already set the iod_size, just curious any test fails without this patch?

Reply:
hmm, args->iods has been changed in obj_rw_req_reassemb() ? yes, daos io test will fail on EC object.

Reply:
are you reworking it?

Reply:
oh, no, I think the patch is correct. xuezhao?

Reply:
ok, seems fine. probably some codes can be cleanup later on following patch.